tests/test_synthetic_volume.py

```diff
@@ -0,0 +1,72 @@
+from unittest import TestCase
+
+import numpy as np
+from parameterized import parameterized_class
+
+from aspire.source.simulation import Simulation
+from aspire.volume import (
+    AsymmetricVolume,
+    CnSymmetricVolume,
+    DnSymmetricVolume,
+    LegacyVolume,
+    OSymmetricVolume,
+    TSymmetricVolume,
+)
+
+
+class Base:
+    def setUp(self):
+        self.dtype = np.float32
+        self.L = 10
+        self.C = 1
+        self.seed = 0
+        vol_kwargs = dict(
+            L=self.L,
+            C=self.C,
+            seed=self.seed,
+            dtype=self.dtype,
+        )
+        if hasattr(self, "order"):
+            vol_kwargs["order"] = self.order
+
+        self.vol = self.vol_class(**vol_kwargs)
+
+    def testVolumeRepr(self):
+        """Test Synthetic Volume repr"""
+        self.assertTrue(repr(self.vol).startswith(f"{self.vol.__class__.__name__}"))
+
+    def testVolumeGenerate(self):
+        """Test that a volume is generated"""
+        _ = self.vol.generate()
+
+    def testSimulationInit(self):
+        """Test that a Simulation initializes provided a synthetic Volume."""
+        _ = Simulation(L=self.L, vols=self.vol.generate())
+
+
+@parameterized_class(("order"), [(3,), (4,), (5,), (6,)])
+class CnSymmetricVolumeCase(Base, TestCase):
+    vol_class = CnSymmetricVolume
+    order = 2
+
+
+@parameterized_class(("order"), [(3,), (4,), (5,), (6,)])
+class DnSymmetricVolumeCase(Base, TestCase):
```

Comment:
Where are you testing the Dn volume is Dn? Same for T and O etc. Cn, legacy, and asym appear to be in the volume tests and can live here.  Let's try to have `test_volume` focus on the underlying `Volume` class... and this one on the synthetic generators.

This is probably a natural place for a `test_volume()` or `test_symmetry()` method or something to that effect.

If you move the `self.vol.generate()` into setUp you can avoid regenerating the volume.

Reply:
As I mentioned, if you prefer to make an issue and come back to this so the code can move forward I would be fine with that (but you'll be working on that issue next...).

Reply:
created issue #758